.github/workflows/ci.yml

```
@@ -0,0 +1,82 @@
+
+jobs:
+  semgrep:
+    name: Run Semgrep Scan
+    runs-on: ubuntu-latest
+    # 設定此 job 失敗時仍然不阻斷整個 workflow
+    continue-on-error: true
+    outputs:
+      semgrep_results: ${{ steps.check.outputs.semgrep_results }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)